lib/src/spec/common.dart

```diff
@@ -398,17 +442,31 @@ abstract class EventEmitter<E, G> {
   Stream<G> on([E event]);
 }
 
-abstract class PaginatedResult<T> {
-  List<T> items();
-  Future<PaginatedResult<T>> first();
-  Future<PaginatedResult<T>> next();
-  Future<PaginatedResult<T>> current();
+/// PaginatedResult [TG1](https://docs.ably.io/client-lib-development-guide/features/#TG1)
+///
+/// A type that represents page results from a paginated query.
+/// The response is accompanied by metadata that indicates the
+/// relative queries available.
+abstract class PaginatedResultInterface<T> {
+  /// items contain page of results (TG3)
+  List<T> items;
+
+  /// returns a new PaginatedResult loaded with the next page of results. (TG4)
+  ///
+  /// If there are no further pages, then null is returned
+  Future<PaginatedResultInterface<T>> next();
+
+  /// returns a new PaginatedResult for the first page of results (TG5)
+  Future<PaginatedResultInterface<T>> first();
+
+  /// returns true if there are further pages (TG6)
   bool hasNext();
+
+  /// returns true if this page is the last page (TG7)
   bool isLast();
 }
 
-abstract class HttpPaginatedResponse extends PaginatedResult<String> {
-  List<String> items();
+abstract class HttpPaginatedResponse extends PaginatedResultInterface<String> {
```

Comment:
[same as another comment] We are exporting all the definitions in the final package `ably/ably.dart` which is leading to duplicate declarations.

Reply:
But `Interface` is misleading then. `Base` might be a better extension then.

Reply:
Do they have the same name in other implementations?

Reply:
@zoechi I created these base classes (spec) so that even if we migrate from a plugin based approach to a pure dart package, we should be able to do it smoothly.

And currently: whenever it is required to implement any of these classes, renaming the class in spec with `XXXImplementation` and then implementing a platform object for the same is serving the purpose.

What do you think would be the best approach / suggestion for modification in relation to above comments?